### PR TITLE
Port to dune and generate better docs

### PR DESCRIPTION
**Pull request by mseri**
This will fully work only once https://github.com/ocaml/opam-repository/pull/13043 is merged

### CODE REVIEW

**Comment by mseri** [comment]
I have already pushed the generated documentation to http://mirage.github.io/ocaml-rpc/rpclib/

File: rpclib.opam [hunk 1]
```diff
@@ -1,25 +1,23 @@
...
   "xmlm"
   "yojson"
 ]
-available: [ ocaml-version >= "4.04.0" ]
-synopsis: "A library to deal with RPCs in OCaml"
\ No newline at end of file
+flags: light-uninstall
+build: [
+  ["dune" "build" "-p" name "-j" jobs]
+  ["dune" "runtest" "-p" name] {with-test}
```
Comment (gaborigloi) [inline rpclib.opam]: We can use `run-test` instead

**Comment by gaborigloi** [comment]
Could we put the index.html into the directory root?
http://mirage.github.io/ocaml-rpc/index.html

**Comment by mseri** [comment]
Not for the moment. If it will become customisable in `dune` or `odoc` I will submit a new PR

File: rpclib.opam (hunk 1 above)
Comment (mseri) [reply rpclib.opam]: Most new packages in opam (and the opam automated conversion tools) use `{with-test}`, I decided to stick with that. Moreover, this is what was spit out by this script: https://gist.github.com/mseri/b0d28a9c56303ac17580da9b494a68ce (I did not want to do the conversion by hand :P)

**Comment by mseri** [comment]
@gaborigloi I have pushed to gh-pages an index file for the root that links to the README generated documentation and to the rpclib subfolder. I don't know how long it will take before it becomes visible.

**Comment by mseri** [comment]
(it's already there, you just have to force the cdn to refresh the cache: http://mirage.github.io/ocaml-rpc/?version=foobar)

**Pull request merged**